custom_components/tahoma/climate_devices/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,160 @@
+    OverkizCommandParam.ECO: OverkizCommand.SET_ECO_TEMPERATURE,
+    OverkizCommandParam.SECURED: OverkizCommand.SET_SECURED_POSITION_TEMPERATURE,
+}
+
+
+class SomfyHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Heating Temperature Interface."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_supported_features = SUPPORT_PRESET_MODE
+
+    def __init__(self, device_url: str, coordinator: OverkizDataUpdateCoordinator):
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation i.e. heat, cool mode."""
+        if (
+            self.executor.select_state(OverkizState.CORE_ON_OFF)
+            == OverkizCommandParam.OFF
+        ):
+            return HVAC_MODE_OFF
+
+        state = self.executor.select_state(
+            OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_ACTIVE_MODE
+        )
+        if state not in OVERKIZ_TO_HVAC_MODES:
+            if state is not None:
+                # Unknown and potentially a new state, log to make it easier to report
+                _LOGGER.error("Overkiz state unknown: %s", state)
+            return HVAC_MODE_OFF
+
+        return OVERKIZ_TO_HVAC_MODES[state]
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_ACTIVE_MODE, HVAC_MODES_TO_OVERKIZ[hvac_mode]
+        )
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return OVERKIZ_TO_PRESET_MODES[
+            self.executor.select_state(
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_SETPOINT_MODE
+            )
+        ]
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_MANU_AND_SET_POINT_MODES,
+            PRESET_MODES_TO_OVERKIZ[preset_mode],
+        )
+
+    @property
+    def hvac_action(self) -> Optional[str]:
+        """Return the current running hvac operation if supported."""
+        current_operation = self.executor.select_state(
+            OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_OPERATING_MODE
+        )
+
+        if current_operation not in OVERKIZ_TO_HVAC_ACTION:
+            if current_operation is not None:
+                # Unknown and potentially a new state, log to make it easier to report
+                _LOGGER.error("Overkiz state unknown: %s", current_operation)
+            return None
+
+        return OVERKIZ_TO_HVAC_ACTION[current_operation]
```

Comment:
Same comment that https://github.com/iMicknl/ha-tahoma/pull/703/files#r784721031